### PR TITLE
Improve metrics

### PR DESCRIPTION
**Pull request by CyrusRoshan**
Host metrics working
Wildcard metrics working

### CODE REVIEW

File: pkg/agent/metrics/metrics.go [hunk 1]
```diff
@@ -9,24 +9,48 @@ import (
...
-		"time",
-		"netdev",
-	)
+var (
+	ErrInvalidCollector = errors.New("invalid collector")
+
+	DefaultNodeCollectorConfig = NodeCollectorConfig{
+		ProcFSPath: "/host/proc",
+		SysFSPath:  "/host/sys",
+		RootFSPath: "/",
```
Comment (joshwget) [inline pkg/agent/metrics/metrics.go]: Should this be `/host`? Seems weird that the other defaults have `/host` but not this one.

File: pkg/agent/service/service.go [hunk 2]
```diff
@@ -18,6 +18,17 @@ import (
 	gossh "golang.org/x/crypto/ssh"
 )
 
+const (
+	HostMetricsBrokenMsg = "host metrics are not working, check agent logs for details"
+)
+
+var (
+	hostMetricsFallbackHandler = http.Handler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
```
Comment (joshwget) [inline pkg/agent/service/service.go]: `http.Error` is good for this use case

File: pkg/agent/service/service.go [hunk 3]
```diff
@@ -42,19 +53,18 @@ func NewService(
 	}
 	go s.getSigner()
 
+	hostMetricsHandler, err := metrics.HostMetricsHandler(nil)
+	if err != nil {
+		log.WithError(err).Error("create host metrics handler")
+		hostMetricsHandler = &hostMetricsFallbackHandler
```
Comment (joshwget) [inline pkg/agent/service/service.go]: Awesome 👍

File: pkg/controller/runner/datadog/filter.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import (
+	"fmt"
+
+	"github.com/deviceplane/deviceplane/pkg/metrics/datadog"
+	"github.com/deviceplane/deviceplane/pkg/models"
+)
+
+const WildcardMetric = string("*")
+
+func FilterMetrics(
```
Comment (joshwget) [inline pkg/controller/runner/datadog/filter.go]: Def makes sense to have this in its own file 👍 

The logic is starting to get complex enough here to justify unit tests

File: pkg/controller/service/service.go [hunk 5]
```diff
@@ -1098,27 +1098,31 @@ func (s *Service) createProject(w http.ResponseWriter, r *http.Request, authenti
 		return
 	}
 
+	allowAllMetrics := models.MetricConfig{
+		Metrics: []models.Metric{
+			{
+				Metric: "*",
+				Labels: []string{},
+				Tags:   []string{"device"},
```
Comment (joshwget) [inline pkg/controller/service/service.go]: This is a pretty high cardinality default - probably better to have this turned off initially

File: pkg/utils/utils.go [hunk 6]
```diff
@@ -25,6 +29,8 @@ func ProxyResponse(w http.ResponseWriter, resp *http.Response) {
 			w.Header().Add(key, value)
 		}
 	}
+	w.Header().Set(ProxiedFromDeviceHeader, "")
```
Comment (joshwget) [inline pkg/utils/utils.go]: Can you check that this function is only used in the controller? I think the agent might also call it

File: ui/src/config.js [hunk 7]
```diff
@@ -16,7 +16,7 @@ const config = process.env.REACT_APP_ENVIRONMENT === 'development'
   : production;
 
 export default {
-  agentVersion: '1.7.3',
+  agentVersion: '1.7.5',
```
Comment (joshwget) [inline ui/src/config.js]: Let's move to separate PR/commit. It'd be weird for this to end up in the squashed commit for this PR.

File: ui/src/App.js [hunk 8]
```diff
@@ -648,9 +651,9 @@ class InnerOogie extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      tabs: ['devices', 'provisioning', 'applications', 'iam'],
-      tabLabels: ['Devices', 'Provisioning', 'Applications', 'IAM'],
-      icons: ['desktop', 'box', 'application', 'user'],
+      tabs: ['devices', 'provisioning', 'applications', 'metrics', 'iam'],
+      tabLabels: ['Devices', 'Provisioning', 'Applications', 'metrics', 'IAM'],
+      icons: ['desktop', 'box', 'application', 'key', 'user'],
```
Comment (joshwget) [inline ui/src/App.js]: `vertical-bar-chart-asc` icon would be better

File: ui/src/App.js [hunk 9]
```diff
@@ -130,6 +131,8 @@ class AddDevice extends Component {
         "-v /var/lib/deviceplane:/var/lib/deviceplane",
         "-v /var/run/docker.sock:/var/run/docker.sock",
         "-v /etc/os-release:/etc/os-release",
+        "-v /sys:/host/sys",
```
Comment (joshwget) [inline ui/src/App.js]: Let's also move this out into a follow up PR.

Comment (joshwget) [reply ui/src/App.js]: `-v "/:/host:ro,rslave"` (from node_exporter docs) is probably a better choice than these two independent bind mounts as well.

File: pkg/agent/metrics/metrics.go (hunk 1 above)
Comment (CyrusRoshan) [reply pkg/agent/metrics/metrics.go]: Sure, I don't really know much about rootfs, honestly

File: pkg/controller/runner/datadog/filter.go (hunk 4 above)
Comment (CyrusRoshan) [reply pkg/controller/runner/datadog/filter.go]: Ah yes, the metric system, unit tests. There's a joke in here somewhere.

File: pkg/controller/service/service.go (hunk 5 above)
Comment (CyrusRoshan) [reply pkg/controller/service/service.go]: This is only applied to state metrics. I think it'll be fine. It's just one metric per device, which makes total sense to want to see when you slap in your DD key. Product side choice here.

File: pkg/utils/utils.go (hunk 6 above)
Comment (CyrusRoshan) [reply pkg/utils/utils.go]: It is, but would that matter?

File: ui/src/App.js (hunk 8 above)
Comment (CyrusRoshan) [reply ui/src/App.js]: no man, you keep on saying, metrics are _KEY_

Comment (CyrusRoshan) [reply ui/src/App.js]: jk yeah sure lol

File: ui/src/App.js (hunk 9 above)
Comment (CyrusRoshan) [reply ui/src/App.js]: So basically, just move bumping the version in the UI to a separate PR? down.

File: pkg/utils/utils.go (hunk 6 above)
Comment (joshwget) [reply pkg/utils/utils.go]: Not other than the header being set somewhere where it's not really being proxied. We could always just make a second function with this header and then use that in the controller only.

File: pkg/controller/service/service.go (hunk 5 above)
Comment (joshwget) [reply pkg/controller/service/service.go]: Okay for now, but I do worry about this down the line. Let's just make a note to rethink later.

File: pkg/utils/utils.go (hunk 6 above)
Comment (CyrusRoshan) [reply pkg/utils/utils.go]: In both cases it still comes from the device, so it sounds reasonable to set it at both locations, but I've got no strong opinion, so I'll defer on this
